src/NzbDrone.Core/Extras/Metadata/MetadataService.cs

```diff
@@ -269,15 +268,30 @@ private List<MetadataFile> ProcessMovieImages(IMetadata consumer, Movie movie, L
 
                 _otherExtraFileRenamer.RenameOtherExtraFile(movie, fullPath);
 
-                var metadata = GetMetadataFile(movie, existingMetadataFiles, c => c.Type == MetadataType.MovieImage &&
-                                                                                   c.RelativePath == image.RelativePath) ??
+                var hash = image.Url.SHA256Hash();
```

Comment:
I need to include all this code to avoid old files remain.

If you have a movie like "MovieTitle-720p-poster.jpg" and you upgrade to for example 1080p, the new and the old file name are different and Radarr doesn't know that the first has to be removed. 

The use of hash implies that all the old Images must be recreated to obtain and set the hash in database. Maybe with one CleanImage Event when user updates to the app version that includes this change?